test/endpoints/automations.test.js

```diff
@@ -112,6 +139,62 @@ describe('Tests for automation endpoints\n', () => {
         done();
       });
   });
+  it('Should filter successful freckleAutomations data', (done) => {
+    chai
+      .request(app)
+      .get('/api/v1/automations?page=1&limit=5&freckleAutomation=success')
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        expect(res.body)
+          .to.have.property('message')
+          .to.equal('Successfully fetched automations');
+        expect(res.body)
+          .to.have.property('data')
+          .to.have.lengthOf.below(10);
+        expect(res.body.data[0].freckleAutomations.status)
+          .to.be.equal('success');
+        done();
+      });
+  });
+  it('Should filter successful slackAutomations data', (done) => {
+    chai
+      .request(app)
+      .get('/api/v1/automations?page=1&limit=5&slackAutomation=success')
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        expect(res.body)
+          .to.have.property('message')
+          .to.equal('Successfully fetched automations');
+        expect(res.body.data[0].slackAutomations.status)
+          .to.be.equal('success');
+        done();
+      });
+  });
+  it('Should filter successful emailAutomations data', (done) => {
+    chai
+      .request(app)
+      .get('/api/v1/automations?page=1&limit=5&emailAutomation=success')
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        expect(res.body)
+          .to.have.property('message')
+          .to.equal('Successfully fetched automations');
+        expect(res.body.data)
+          .to.have.lengthOf.below(10);
```

Comment:
length below 10 can mean 0 which will allow the test to pass

Reply:
@Nerldy can you update this to reflect the number of email success count base on the generated automation like how we did for date query.  Filter out all automation object that has related email automation objects to be failures and keep those with related email automation objects with status success then use this automation objects collection to assert with the response. It might require you add email automation object collections with a status of both success and failure to the generated automation collection (Do the same for slack and freckle).